src/middlewared/middlewared/plugins/jail.py

```diff
@@ -426,21 +426,17 @@ def fstab(self, jail, options):
         destination = options.get('destination')
         if destination:
             destination = f'/{destination}' if destination[0] != '/' else destination
-            dst = f'{self.get_iocroot()}/jails/{jail}/root{destination}'
-            if 'mnt' in destination:
-                verrors.add(
-                    'destination',
-                    'Destination should not include the jail path. '
-                    'e.g "/mnt/iocage/jails/btsync/root/" Path should be specified'
-                    'after /'
-                )
-            elif os.path.exists(dst):
-                if not os.path.isdir(dst):
+            dst = f'{self.get_iocroot()}/jails/{jail}/root'
+            if not dst in destination:
+                destination = f'{dst}{destination}'
```

Comment:
You're passing the full jail path to iocage fstab, that was not the case before, are you sure thats right?